Spigot-Server-Patches/0444-Add-per-world-spawn-limits.patch

```diff
@@ -0,0 +1,112 @@
+From ed89877cc0b6e209b784ed88cc468683588c64e5 Mon Sep 17 00:00:00 2001
+From: chase <chasewhip20@gmail.com>
+Date: Sun, 23 Feb 2020 01:03:27 -0700
+Subject: [PATCH] Add per world spawn limits Adds a config option to support
+ moddifying spawn limits per world.
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index 7d408542e..78d5c9612 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -8,6 +8,7 @@ import java.util.Map;
+ 
+ import com.destroystokyo.paper.antixray.ChunkPacketBlockControllerAntiXray.ChunkEdgeMode;
+ import com.destroystokyo.paper.antixray.ChunkPacketBlockControllerAntiXray.EngineMode;
++import net.minecraft.server.EnumCreatureType;
+ import net.minecraft.server.MinecraftServer;
+ import org.bukkit.Bukkit;
+ import org.bukkit.Material;
+@@ -658,4 +659,44 @@ public class PaperWorldConfig {
+     private void nerfNetherPortalPigmen() {
+         nerfNetherPortalPigmen = getBoolean("game-mechanics.nerf-pigmen-from-nether-portals", nerfNetherPortalPigmen);
+     }
++
++
++    public Map<EnumCreatureType, Integer> spawnLimitOverride = new java.util.HashMap<>();
++    private void worldSpawnLimitOverride() {
++        Map<String, Integer> example = new HashMap<>();
++        for (EnumCreatureType creatureType : EnumCreatureType.values()){
++            if (creatureType == EnumCreatureType.MISC)
++                continue;
++            example.put(creatureType.a(), 5);
++        }
++        config.addDefault("world-spawn-limit-override.example", example);
++
++        ConfigurationSection limits = config.getConfigurationSection("world-spawn-limit-override." + worldName);
++        if (limits != null) {
++            for (String key : limits.getKeys(false)) {
++                EnumCreatureType creatureType = null;
++                for (EnumCreatureType type : EnumCreatureType.values()){
++                    if (type.a().equals(key)){
++                        creatureType = type;
++                    }
++                }
++
++                if (creatureType == null)
++                    continue;
++
++                String limitString = limits.getString(key);
++                int limit;
++                try {
++                    limit = Integer.valueOf(limitString);
++                } catch (Exception e) {
++                    limit = limitString.hashCode();
```

Comment:
What's this? Shouldn't an error message and a fallback value of -1 be better? Also, it doesn't really change anything, but only NumberFormatException should be caught.